On Ventura, exclusively use GameController.framework for gamepads (disable WebKit's IOHID code path)

beidson opened the pull request:
#### 26a4b6d1625f7178f48c3cd34382c2e6bae67903
<pre>
On Ventura, exclusively use GameController.framework for gamepads (disable WebKit&apos;s IOHID code path)
<a href="https://bugs.webkit.org/show_bug.cgi?id=248136">https://bugs.webkit.org/show_bug.cgi?id=248136</a>
rdar://102264246

Reviewed by Darin Adler.

On Ventura, GameController framework supports every controller we had interest in supporting
via our custom IOHID code.

So let&apos;s *only* use the GameController framework code path on Ventura. The support will be superior!

Also fixes some API tests on Ventura, as Ventura also introduced some race conditions between
the IOHID and GameController code paths.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/Modules/gamepad/GamepadManager.cpp:
(WebCore::GamepadManager::makeGamepadVisible):
(WebCore::GamepadManager::registerNavigator):
(WebCore::GamepadManager::unregisterNavigator):
(WebCore::GamepadManager::registerDOMWindow):
(WebCore::GamepadManager::unregisterDOMWindow):
(WebCore::GamepadManager::maybeStartMonitoringGamepads):
(WebCore::GamepadManager::maybeStopMonitoringGamepads):
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.mm:
(WebCore::disableDefaultSystemAction):
* Source/WebCore/platform/gamepad/cocoa/GameControllerSoftLink.mm:
* Source/WebKit/UIProcess/Gamepad/cocoa/UIGamepadProviderCocoa.mm:
* Tools/TestWebKitAPI/Tests/mac/HIDGamepads.mm:

Canonical link: <a href="https://commits.webkit.org/256918@main">https://commits.webkit.org/256918@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/957f8b44af72ad0d1984e79bbf5c0405076b911a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97219 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6486 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30362 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106738 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/167007 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101188 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6771 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35219 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89614 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103424 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102885 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5093 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83846 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32070 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/86943 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88766 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/75007 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/88134 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/507 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/83580 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/490 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21693 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/28454 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4768 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5290 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44188 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/86289 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1731 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41015 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/19433 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/Modules/gamepad/GamepadManager.cpp:
```diff
@@ -131,7 +131,7 @@ void GamepadManager::platformGamepadInputActivity(EventMakesGamepadsVisible even
 void GamepadManager::makeGamepadVisible(PlatformGamepad& platformGamepad, HashSet<NavigatorGamepad*>& navigatorSet, HashSet<DOMWindow*>& domWindowSet)
 {
-    LOG(Gamepad, "GamepadManager::makeGamepadVisible - New gamepad '%s' is visible", platformGamepad.id().utf8().data());
+    LOG(Gamepad, "(%u) GamepadManager::makeGamepadVisible - New gamepad '%s' is visible", getpid(), platformGamepad.id().utf8().data());
```
Since `getpid` returns `pid_t` and not `int` or `unsigned` this needs a cast to `unsigned` or some other technique.

beidson committed:
On Ventura, exclusively use GameController.framework for gamepads (disable WebKit's IOHID code path)
https://bugs.webkit.org/show_bug.cgi?id=248136
rdar://102264246

Reviewed by Darin Adler.

On Ventura, GameController framework supports every controller we had interest in supporting
via our custom IOHID code.

So let's *only* use the GameController framework code path on Ventura. The support will be superior!

Also fixes some API tests on Ventura, as Ventura also introduced some race conditions between
the IOHID and GameController code paths.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/Modules/gamepad/GamepadManager.cpp:
(WebCore::GamepadManager::makeGamepadVisible):
(WebCore::GamepadManager::registerNavigator):
(WebCore::GamepadManager::unregisterNavigator):
(WebCore::GamepadManager::registerDOMWindow):
(WebCore::GamepadManager::unregisterDOMWindow):
(WebCore::GamepadManager::maybeStartMonitoringGamepads):
(WebCore::GamepadManager::maybeStopMonitoringGamepads):
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.mm:
(WebCore::disableDefaultSystemAction):
* Source/WebCore/platform/gamepad/cocoa/GameControllerSoftLink.mm:
* Source/WebKit/UIProcess/Gamepad/cocoa/UIGamepadProviderCocoa.mm:
* Tools/TestWebKitAPI/Tests/mac/HIDGamepads.mm:

Canonical link: https://commits.webkit.org/256918@main
